src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -343,6 +346,10 @@ public RevisionsCollectionConfiguration GetRevisionsConfiguration(string collect
             if (configuration == null)
                 configuration = GetRevisionsConfiguration(collectionName.Name);
 
+            if (configuration.Disabled && 
+                nonPersistentFlags.Contain(NonPersistentDocumentFlags.FromReplication) == false)
+                return false;
```

Comment:
How is that change related to introduced option of defining separate conflict revision configuration? Or this is another fix?

Reply:
I guess it both.

On conflict we were passing the configuration directly, and if we don't check for `Disabled` we will keep conflicted revision even if the they were disabled.
In general if the configuration is disabled we should simply leave the state as is.